services/JsonReader.class.php

```diff
@@ -0,0 +1,344 @@
+                        throw new Exception("Cannot Add Multiple Objects To Non-Array Type ".get_class ($val).var_dump($val));
+                    }
+                }
+                **/
+            array_splice($pathArray, 0, 1);
+            $newKey = implode(";", $pathArray);
+            self::addToJsonMap($arr[$subKey], $newKey, $value);
+
+        }
+    }
+
+    /**
+     * Function: delete an Key -> Value from @param arr
+     * @param arr The Object we are deleting from.
+     * @param key prefix + keyName sperated by dilimiter ';'
+     *        @param prefix the path of the object you want to put `keyName` under
+     *            Ex using the Json Reference object above
+     *               if you want to add ['phoneNumber'] => ['2020020202'] under `indexes`
+     *               prefix = `data;indexes;' and key = `phoneNumber`
+     *              * Notice Prefix end has ;
+     *            Ex adding to the main object
+     *              prefix =  "" key = "phoneNumber"
+     *
+     */
+    function deleteFromJsonMap(&$arr, $key)
+    {
+        $pathArray = explode(";", $key);
+        $size = sizeof($pathArray);
+        print ("Size of Key to delete is: '" . $key . "' is " . $size . "\n");
+        var_dump($pathArray);
+        if (sizeof($pathArray) == 1)
+        {
+            // No need to traverse
+            settype($arr, "array");
+
+            if (!isset($arr[$key]))
+            {
+                echo "4";
+                return new Exception("Trying To Delete Something That Does Not Exist");
+                echo "3";
+            }
+            echo "2";
+            //Not sure if this is working
+            unset($arr[$key]);
+            var_dump($arr);
+            return;
+        }
+        else
+        {
+            print ("<br>MultiLine Key<br>");
+            $subKey = $pathArray[0];
+            $val = $arr[$subKey];
+            //if val not set, it means the FULL path does not exist
+            if (!isset($val))
+            {
+                throw new Exception("Invalid Path to delete");
+            }
+            array_splice($pathArray, 0, 1);
+            $newKey = implode(";", $pathArray);
+            self::deleteFromJsonMap($arr[$subKey], $newKey);
+        }
+    }
+
+    /**
+     * Function: retrive a Value based on Key from @param arr
+     * @param arr Object that is being added to or being manipulated
+     *
+     * @param key prefix + keyName sperated by dilimiter ';'
+     *        @param prefix the path of the object you want to put `keyName` under
+     *            Ex using the Json Reference object above
+     *               if you want to add ['phoneNumber'] => ['2020020202'] under `indexes`
+     *               prefix = `data;indexes;' and key = `phoneNumber`
+     *              * Notice Prefix end has ;
+     *            Ex adding to the main object
+     *              prefix =  "" key = "phoneNumber"
+     */
+    function fetchFromJsonMap($arr, $key)
+    {

```

Comment:
the methods fetchFromJsonMap, deleteFromJsonMap, and addToJsonMap are doing similar things in the beginning: explode, if size ==1.
I did not look at the functionality in very much detail, but if a helper method can be made, please do so. 
If these methods are completely different then keep it as is